synapse_domain_rule_checker/__init__.py

```diff
@@ -31,19 +33,41 @@ class Membership:
 @attr.s(auto_attribs=True, frozen=True)
 class DomainRuleCheckerConfig:
     can_invite_if_not_in_domain_mapping: bool
-    domain_mapping: Optional[Dict[str, List[str]]] = None
+    use_regex: bool = False
+    domain_mapping: Dict[str, List[str]] = dict()
     can_only_join_rooms_with_invite: bool = False
     can_only_invite_during_room_creation: bool = False
     can_invite_by_third_party_id: bool = True
-    domains_prevented_from_being_invited_to_published_rooms: Optional[List[str]] = None
+    domains_prevented_from_being_invited_to_published_rooms: List[str] = []
+
+    def domain_map(self) -> Dict[Pattern[str], Set[Pattern[str]]] | Dict[str, Set[str]]:
```

Comment:
Unfortunatly, the `|` syntax for unions was only added in Python 3.10 (see https://peps.python.org/pep-0604/ and https://docs.python.org/3/whatsnew/3.10.html)